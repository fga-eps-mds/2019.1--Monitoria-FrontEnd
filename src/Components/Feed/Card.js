import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import classnames from 'classnames';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Collapse from '@material-ui/core/Collapse';
import Avatar from '@material-ui/core/Avatar';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import red from '@material-ui/core/colors/red';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import SvgIcon from '@material-ui/core/SvgIcon'
import { Grid } from "@material-ui/core";
import axios from 'axios';

function Add(props) {
  return (
    <SvgIcon {...props}>
      <svg>
        <path fill="#000000" d="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z" />
      </svg>
    </SvgIcon>
  );
}

const styles = theme => ({
  card: {
    minWidth: '100%',
  },
  media: {
    height: 0,
    paddingTop: '56.25%', // 16:9
  },
  actions: {
    display: 'flex',
  },
  expand: {
    transform: 'rotate(0deg)',
    marginLeft: 'auto',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  },
  avatar: {
    backgroundColor: red[500],
  },
});

class RecipeReviewCard extends React.Component {
  
 
    state =  {lista : [],name_monitoring:'', matter:'', expanded: false}
    async componentDidMount() {
     await axios.get(`https://gist.githubusercontent.com/caiooliv/17243478be5bddb4f26fcba90b25a031/raw/fa9049b4ae9aa6d283edadd0d2e2e1ec81cb81bc/teste.json`)
      .then(res => {
        console.log('entrou')
        const person = res.data
        this.setState({name_monitoring:person['name_monitoring'], matter: person['matter'] })
      })
  }


  handleExpandClick = () => {
    this.setState(state => ({ expanded: !state.expanded }));
  };

  render() {
    const { classes } = this.props;

    return (
      <Card className={classes.card}>
      <Grid container
  direction="column"
  justify="center"
  alignItems="center">
  
        <CardHeader
         
          title={this.state.name_monitoring}
          subheader={this.state.matter}
          
        />
        </Grid>
        <CardActions className={classes.actions} disableActionSpacing>
          <IconButton aria-label="Add to favorites">
            <Add />
          </IconButton>
          <IconButton
            className={classnames(classes.expand, {
              [classes.expandOpen]: this.state.expanded,
            })}
            onClick={this.handleExpandClick}
            aria-expanded={this.state.expanded}
            aria-label="Mostrar Mais"
          >
            <ExpandMoreIcon />
          </IconButton>
        </CardActions>
        <Collapse in={this.state.expanded} timeout="auto" unmountOnExit>
          <CardContent>
            <Typography paragraph>Descrição:</Typography>
            <Typography paragraph>
              Lorem Ipsum 
            </Typography>
          </CardContent>
        </Collapse>
      </Card>
    );
  }
}

RecipeReviewCard.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(RecipeReviewCard);