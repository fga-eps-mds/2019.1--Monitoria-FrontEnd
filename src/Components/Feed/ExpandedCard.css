.profilePicture{
    width: 120;
    height:90;
}

.teste{
    background-color: #eeeeee;
}

.div {
    overflow: hidden;
  }