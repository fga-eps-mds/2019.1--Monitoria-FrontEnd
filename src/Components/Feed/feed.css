.divAdd{
    position: relative;
    bottom: 10%;
    right:0;
    padding: 24000;
}