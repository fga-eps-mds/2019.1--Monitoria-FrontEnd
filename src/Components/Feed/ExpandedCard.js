import React from 'react';
import { Grid, Typography,Button } from "@material-ui/core";
import firebase from 'firebase';
import axios from 'axios';
import AppBar from './AppBarWithBack';
import './ExpandedCard.css'


import { ReactComponent as Logo } from '../../Assets/svg/telegram.svg';
import { ReactComponent as Like } from '../../Assets/svg/like.svg';
import Fab from '@material-ui/core/Fab';
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';
import {Link} from 'react-router-dom';

const theme = createMuiTheme({
    palette: {
      primary: { main: "#44a1f2" },
      secondary: { main: '#11cb5f' },
    },
    typography: { useNextVariants: true },
});

class ExpandedCard extends React.Component {
    
    state = {
        tutoringName: '',
        tutoringDescription: '',
        tutoringTheme: '',
        monitorName: '',
        photo: '',
        telegram:'',
    }

    componentDidMount() {
        var token = {};
        var idTutoring = this.props.match.params.id_tutoring;
        
        firebase.auth().onAuthStateChanged(user =>{
            this.setState({isSignedIn: !!user});
            if(user){
                firebase.auth().currentUser.getIdToken().then(function(idToken){
                    token["access_token"] = idToken;
                    token["id_tutoring_session"] = idTutoring;
                });
                
                axios.post(process.env.REACT_APP_GATEWAY+"/get_tutoring/", token)
                    .then(res => {
                        const person = res.data
                        this.setState({tutoringName:person["name"], tutoringTheme:person["subject"], tutoringDescription:person["description"],
                                      monitorName: person.monitor["name"], photo:person.monitor["photo"], telegram:person.monitor["telegram"]}) 
                        
                    });
            }
        });
      
          
    }

    
        
  render() {

    var texto =  this.state.telegram;
    var er = texto;
    texto = er.replace('@','');
    console.log(texto);   
        
    var photoUrl = this.state.photo;
    if( photoUrl != null ){
        photoUrl = photoUrl.replace("api-monitoria","localhost")
      } else {
        photoUrl = "https://cdn-eleicoes.gazetadopovo.com.br/fotos/sao-paulo/deputado-federal/batore-1444.jpg"
      }

    return (
        <div className='div'>
            
            <Grid style={{position: "absolute"}} container justify="center" alignItems="stretch">
               <AppBar/>    
            </Grid>            
            <Grid container justify="center" direction="row" style={{paddingTop:80, paddingLeft:15,paddingBottom:15}} className="teste" >              
                <Grid item>                            
                    <img src={photoUrl} style={{width:140, height:140}}></img>
                </Grid>
                <Grid item>    
                    <Grid container justify="center" direction="column" alignItems="center" alignContent="center">
                        <Grid item style={{marginTop:50}} style={{marginLeft:50}} >
                            <h1>Monitor</h1>
                        </Grid>
                        <Grid item>
                            <Typography style={{marginLeft:50}}>
                                Nome: {this.state.monitorName}
                            </Typography>
                        </Grid>
                        <Grid container justify="center" direction="column" alignItems="center" alignContent="center" >
                                                    
                        </Grid>
                    </Grid>
                </Grid>
            </Grid>
            <div>
                <Grid item style={{paddingLeft:15}}> 
                    <Grid container direction="column">
                        <Grid item >
                            <h1>Monitoria</h1>
                        </Grid>
                        <Grid item>
                            <h3>Matéria:</h3>
                            <Typography>
                                {this.state.tutoringName}
                            </Typography>
                        </Grid>
                        <Grid item>
                            <h3>Tema:</h3>
                            <Typography>
                                {this.state.tutoringTheme}
                            </Typography>
                        </Grid>
                        <Grid item>
                            <h3>Descrição:</h3>
                            <Typography>
                                {this.state.tutoringDescription}
                            </Typography>
                        </Grid>
                        <Grid container alignContent="center" justify="center" direction="row" spacing={24} alignItems="center" style={{marginTop: 25}} >
                                                 
                        </Grid>
                    </Grid>
                </Grid>
            </div>
              
            <Grid container alignContent="center" justify="center" direction="row" spacing={24} alignItems="center" style={{marginTop: 25}}>
              <Grid item >
                <MuiThemeProvider theme={theme}>
                    <Fab color="primary" aria-label="Edit" >
                        <Like/>
                    </Fab>
                </MuiThemeProvider>
              </Grid>
              <Grid item>
                    <a href={"https://"+"t.me/" + texto}>{
                        <MuiThemeProvider theme={theme}>
                            <Fab color="primary" aria-label="Edit" >
                                <Logo/>
                            </Fab>
                        </MuiThemeProvider>}
                    </a>
              </Grid>
            </Grid>
        </div>
    );
  }
}


export default (ExpandedCard);