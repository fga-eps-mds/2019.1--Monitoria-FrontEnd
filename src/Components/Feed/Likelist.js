import React from 'react';
import { Grid, Typography,Button } from "@material-ui/core";
import firebase from 'firebase';
import axios from 'axios';
import AppBar from './AppBarWithBack';
import './ExpandedCard.css'

import { createMuiTheme } from '@material-ui/core/styles';


import {withRouter} from 'react-router-dom';
import Card from '../Feed/CardLike';


const theme = createMuiTheme({
    palette: {
      primary: { main: "#44a1f2" },
      secondary: { main: '#11cb5f' },
    },
    typography: { useNextVariants: true },
    overrides: {
        MuiButton: {
          raisedPrimary: {
            color: 'white',
          },
        },
    },
});

class LikeList extends React.Component {
 
    state={
      person: [],
      id_user:'',
      likes:[],
      total_likes:0
    }

    componentDidMount=  async () =>{
      var token = {};
      var idTutoring = this.props.match.params.id_tutoring;
      await firebase.auth().onAuthStateChanged(user =>{
        this.setState({isSignedIn: !!user});
            if(user){
                this.setState({id_user:user.uid})
                firebase.auth().currentUser.getIdToken().then(function(idToken){
                    token["access_token"] = idToken;
                    token["id_tutoring_session"] = idTutoring;
                });
                
                axios.post(process.env.REACT_APP_GATEWAY+"/get_tutoring/", token)
                    .then(res => {
                      this.setState({person:res.data});
                      this.setState({ likes:this.state.person.likes,
                      total_likes:this.state.person.total_likes});
                      
                           
                    });
                   
            }else{
                this.props.history.push('/');
            }
        });  
      

      
    
    }
    
    
   
  render() {
      
    

    return (
        <div className='div'>
        <Grid style={{position: "absolute"}} container justify="center" alignItems="stretch">
               <AppBar router={`/expandedCard/${this.props.match.params.id_tutoring}`}/>    
        </Grid>
        <Grid justify="center" direction="column" alignItems="center"  style={{paddingTop:70,marginTop:10, paddingBottom:40}}> 
         
          <h1>{this.state.likes.map(function(item, i){
            return (
                <Grid item key={i} lg={12} sm={12} container style={{paddingBottom:3}} >
                    <Card name_user={item.user_that_likes.name}photo={item.user_that_likes.photo}  />
                </Grid>
            );
          })}</h1>
        </Grid>
      
       </div>
    );
  }
}

LikeList = withRouter(LikeList);
export default (LikeList);