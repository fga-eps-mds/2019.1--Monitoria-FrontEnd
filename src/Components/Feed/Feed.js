import React, { Component } from 'react';
import { Grid } from '@material-ui/core' ;
import AppBar from '../AppBar/AppBar.js';
import { Link } from 'react-router-dom';
import axios from 'axios';
import firebase from 'firebase';
import Spinner from '../Loader/Spinner';

import Card from './Card.js';
import ButtonSizes from '../GenericButtons/Add.js';
import './feed.css';
import Tab from '../Tab/Tab';

import {withRouter} from 'react-router-dom';
import PropTypes from 'prop-types';
import _ from 'lodash';


class TelaFeed extends Component {
    state =  {
        data : [],
        showWarning: false,
        isLoading: false
    }

    componentDidMount() {
        var token = {};

        this.setState({ isLoading: true });
        firebase.auth().onAuthStateChanged(user =>{
            this.setState({isSignedIn: !!user});
            if(user){
                firebase.auth().currentUser.getIdToken().then(function(idToken){
                    token["access_token"] = idToken;
                });
                axios.post(process.env.REACT_APP_GATEWAY+"/all_tutoring/", token)
                    .then(res => {
                        const person = res.data.results
                        this.setState({data:person})
                    });
                this.setState({ isLoading: false });
            }
            else{
                this.props.history.push('/');
            }
        });
    }

  render() {  
    return (
        <div style={{overflowX:'hidden'}} className="FeedBackground">
            <Grid style={{position: "absolute"}} container justify="center" alignItems="stretch">
                <AppBar/>    
            </Grid> 
            <div>
                <Grid container justify="center" direction="column" alignItems="center" spacing={8} style={{paddingTop:70,marginTop:10, paddingBottom:50}}>
                    {_.map(this.state.data, (item, i) => {
                        return (
                            <Grid item key={i} lg={12} sm={12} container style={{paddingBottom:3}} >
                                <Card name_monitoring={item.name} matter={item.subject} 
                                    description={item.description} photo={item.monitor.photo} 
                                    monitorName={item.monitor.name} id_tutoring={item.id_tutoring_session} />
                            </Grid>
                        );
                    })}
                </Grid>
                <Tab ind={0}/>
            </div>
            <Grid item style={{marginTop:225}}>
                {this.state.isLoading ? <Spinner />:
                    <div>
                        <Grid container style={{marginTop:-225}}>
                            <Grid>
                                <ButtonSizes component={Link} to="/Profile"/>
                            </Grid>
                        </Grid>
                    </div>
                }
            </Grid>
        </div>
    );   
  }
}
TelaFeed.propTypes = {
    warning: PropTypes.bool.isRequired,
  };


export default withRouter(TelaFeed);