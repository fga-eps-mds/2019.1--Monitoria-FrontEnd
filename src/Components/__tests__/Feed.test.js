import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import Feed from '../Feed/Feed';

const feed = jest.fn();

test('Logout renders a snapshot properly', () => {
  const tree = renderer.create(
    <Router><Feed feed={feed}/></Router>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});