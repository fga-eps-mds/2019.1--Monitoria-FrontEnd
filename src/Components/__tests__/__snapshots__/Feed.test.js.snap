// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout renders a snapshot properly 1`] = `
<div>
  <div
    className="MuiGrid-container-1 MuiGrid-justify-xs-center-18"
  >
    <div
      className="SimpleAppBar-root-98"
    >
      <div
        className="MuiGrid-container-1"
      >
        <header
          className="MuiPaper-root-108 MuiPaper-elevation4-114 MuiAppBar-root-99 MuiAppBar-positionStatic-103 MuiAppBar-colorPrimary-106"
          style={
            Object {
              "background": "#1DA1F2",
            }
          }
        >
          <div
            className="MuiToolbar-root-135 MuiToolbar-regular-137 MuiToolbar-gutters-136"
          >
            <img
              alt="Logo"
              height="40"
              src="Logo.png"
              width="40"
            />
            <h6
              className="MuiTypography-root-139 MuiTypography-h6-156 MuiTypography-colorInherit-168"
            >
              +Monitoria
            </h6>
          </div>
        </header>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-container-1 MuiGrid-spacing-xs-8-23 MuiGrid-direction-xs-column-4 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18"
    style={
      Object {
        "padding": 80,
      }
    }
  >
    <div
      className="MuiGrid-container-1 MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-true-43"
    >
      <div
        className="MuiPaper-root-108 MuiPaper-elevation1-111 MuiPaper-rounded-109 MuiCard-root-181 RecipeReviewCard-card-175"
      >
        <div
          className="MuiGrid-container-1 MuiGrid-direction-xs-column-4 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18"
        >
          <div
            className="MuiCardHeader-root-182"
          >
            <div
              className="MuiCardHeader-content-185"
            >
              <span
                className="MuiTypography-root-139 MuiTypography-headline-144 MuiCardHeader-title-186"
              >
                Tema: 
              </span>
              <span
                className="MuiTypography-root-139 MuiTypography-body1-148 MuiTypography-colorTextSecondary-172 MuiCardHeader-subheader-187"
              >
                Monitor: 
              </span>
            </div>
          </div>
        </div>
        <div
          className="MuiCardActions-root-188 MuiCardActions-disableActionSpacing-189 RecipeReviewCard-actions-177"
        >
          <button
            aria-label="Add to favorites"
            className="MuiButtonBase-root-197 MuiIconButton-root-191"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiIconButton-label-196"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-200"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <svg>
                  <path
                    d="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
                    fill="#000000"
                  />
                </svg>
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-249"
            />
          </button>
          <button
            aria-expanded={false}
            aria-label="Mostrar Mais"
            className="MuiButtonBase-root-197 MuiIconButton-root-191 RecipeReviewCard-expand-178"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiIconButton-label-196"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-200"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-249"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-container-1"
  >
    <div
      className=""
    >
      <button
        aria-label="Add"
        className="MuiButtonBase-root-197 MuiFab-root-214 MuiFab-secondary-217 MuiFab-sizeMedium-223"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiFab-label-215"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-200"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-249"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-container-1"
  >
    <div
      className=""
    >
      <div
        className="MuiPaper-root-108 MuiPaper-elevation2-112 IconLabelTabs-root-224"
      >
        <div
          className="MuiTabs-root-225"
        >
          <div
            className="MuiTabs-flexContainer-226"
          >
            <div
              className="MuiTabs-scroller-228 MuiTabs-fixed-229"
              onScroll={[Function]}
              role="tablist"
              style={
                Object {
                  "marginBottom": 0,
                }
              }
            >
              <div
                className="MuiTabs-flexContainer-226"
              >
                <a
                  aria-selected={true}
                  className="MuiButtonBase-root-197 MuiTab-root-234 MuiTab-textColorSecondary-238 MuiTab-selected-239 MuiTab-fullWidth-241"
                  href="/Feed"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="tab"
                  tabIndex="0"
                >
                  <span
                    className="MuiTab-wrapper-242"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-200"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <svg>
                        <path
                          d="M10,20V14H14V20H19V12H22L12,3L2,12H5V20H10Z"
                          fill="#1da1f2"
                        />
                      </svg>
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-249"
                  />
                </a>
                <a
                  aria-selected={false}
                  className="MuiButtonBase-root-197 MuiTab-root-234 MuiTab-textColorSecondary-238 MuiTab-fullWidth-241"
                  href="/EditProfile"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="tab"
                  tabIndex="0"
                >
                  <span
                    className="MuiTab-wrapper-242"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-200"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <svg>
                        <path
                          d="M12,4A4,4 0 0,1 16,8A4,4 0 0,1 12,12A4,4 0 0,1 8,8A4,4 0 0,1 12,4M12,14C16.42,14 20,15.79 20,18V20H4V18C4,15.79 7.58,14 12,14Z"
                          fill="#1da1f2"
                        />
                      </svg>
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-249"
                  />
                </a>
                <a
                  aria-selected={false}
                  className="MuiButtonBase-root-197 MuiTab-root-234 MuiTab-textColorSecondary-238 MuiTab-fullWidth-241"
                  href="/Ranking"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="tab"
                  tabIndex="0"
                >
                  <span
                    className="MuiTab-wrapper-242"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-200"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20,2H4V4L9.81,8.36C6.14,9.57 4.14,13.53 5.35,17.2C6.56,20.87 10.5,22.87 14.19,21.66C17.86,20.45 19.86,16.5 18.65,12.82C17.95,10.71 16.3,9.05 14.19,8.36L20,4V2M14.94,19.5L12,17.78L9.06,19.5L9.84,16.17L7.25,13.93L10.66,13.64L12,10.5L13.34,13.64L16.75,13.93L14.16,16.17L14.94,19.5Z"
                        fill="#1da1f2"
                      />
                       
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-249"
                  />
                </a>
                <a
                  aria-selected={false}
                  className="MuiButtonBase-root-197 MuiTab-root-234 MuiTab-textColorSecondary-238 MuiTab-fullWidth-241"
                  href="/Search"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="tab"
                  tabIndex="0"
                >
                  <span
                    className="MuiTab-wrapper-242"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-200"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <svg>
                        <path
                          d="M9.5,3A6.5,6.5 0 0,1 16,9.5C16,11.11 15.41,12.59 14.44,13.73L14.71,14H15.5L20.5,19L19,20.5L14,15.5V14.71L13.73,14.44C12.59,15.41 11.11,16 9.5,16A6.5,6.5 0 0,1 3,9.5A6.5,6.5 0 0,1 9.5,3M9.5,5C7,5 5,7 5,9.5C5,12 7,14 9.5,14C12,14 14,12 14,9.5C14,7 12,5 9.5,5Z"
                          fill="#1da1f2"
                        />
                      </svg>
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-249"
                  />
                </a>
              </div>
              <span
                className="MuiPrivateTabIndicator-root-246 MuiPrivateTabIndicator-colorPrimary-247 MuiTabs-indicator-233"
                style={
                  Object {
                    "left": 0,
                    "width": 0,
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
